### DISCUSSION TITLE
Directed GraphSAGE code sample

### DISCUSSION THREAD

**Discussion by pmoriano** [pyg-team/pytorch_geometric#4113] [Q&A] [answered]
Hello,
Thanks for your hard work and making PyG a first class library for Geometric DL.

I wonder if you can point out to a sample of code implementing and using  graphSAGE for directed graphs. I am unsure on how to  extend the current GraphSAGE layer to support in- and out-neighbors.

Thanks for the help.

**Comment by rusty1s** [comment, marked as answer]
One way to utilize any GNN operator for directed graphs is to utilize two GNN layers for each edge direction. This way, each direction will not share any weights with the reverse one:
```python
x = conv1(x, edge_index).relu() + conv2(x, edge_index.flip([0])).relu()
```
Let me know if that helps you!

**Reply by pmoriano** [reply]
@rusty1s thanks for the quick reply. It helps a lot.
I also found that the issues [#397](https://github.com/pyg-team/pytorch_geometric/issues/397) and [#1845](https://github.com/pyg-team/pytorch_geometric/issues/1845) help to understand the implications of using convolutional layers when dealing with directed graphs.
I hope this maybe of guidance to others.
